### PR TITLE
Add api for autofstrim push and pop

### PR DESCRIPTION
**Pull request by chiachunlin**
Add api for autofstrim to push or pop job from queue.

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by ggriffiths** [comment]
hi @chiachunlin, can you bump the minor version in api.proto and add a changelog note in SDK_CHANGELOG file?

See this recent PR for an example:
https://github.com/libopenstorage/openstorage/commit/ff678099081af0559ad59ef1aabed7af2f82ffe2#diff-5c156a45a762a69cfd45355e32be713830b4f9cf153c86e675cb9e2d38c3f0e7R5

**Comment by chiachunlin** [comment]
> hi @chiachunlin, can you bump the minor version in api.proto and add a changelog note in SDK_CHANGELOG file?
> 
> See this recent PR for an example: [ff67809#diff-5c156a45a762a69cfd45355e32be713830b4f9cf153c86e675cb9e2d38c3f0e7R5](https://github.com/libopenstorage/openstorage/commit/ff678099081af0559ad59ef1aabed7af2f82ffe2#diff-5c156a45a762a69cfd45355e32be713830b4f9cf153c86e675cb9e2d38c3f0e7R5)

Ah, thanks! updated.

**Comment by ggriffiths** [comment]
looks like travis is failing because of the DCO check:
```
git-validation -run DCO,short-subject
 * cda4ce1 "update sdk minor version" ... FAIL
  - FAIL - does not have a valid DCO
 * 49d619e "api for autofstrim push and pop" ... FAIL
  - FAIL - does not have a valid DCO
 ```

@chiachunlin, try squashing your commits and signing the commit with `git commit -s`

File: api/api.proto [hunk 1]
```diff
@@ -1728,48 +1728,65 @@ service OpenStorageFilesystemTrim {
         body: "*"
       };
     }
+
+  rpc AutoFSTrimPush(SdkAutoFSTrimPushRequest)
+      returns (SdkAutoFSTrimPushResponse) {
+    option (google.api.http) = {
+      get : "/v1/filesystem-trim/auto-fstrim-push"
```
Comment (talakad) [inline api/api.proto:1735]: Shouldn't this be put and not get?

File: api/api.proto [hunk 2]
```diff
@@ -1728,48 +1728,65 @@ service OpenStorageFilesystemTrim {
...
-//    `OpenStorageFilesystemCheckClient.Stop()`
-// 7. To Check and Fix errors in the filesystem that are safe to fix, issue a
-//    grpc call to
-//    `OpenStorageFilesystemCheckClient.Start(Mode='fix_safe')`
-//    Status of this operation can be polled in the way mentioned in step 3
-//    This operation can be stopped a Stop request as mentioned in step 6
+  rpc AutoFSTrimPop(SdkAutoFSTrimPopRequest)
+      returns (SdkAutoFSTrimPopResponse) {
+    option (google.api.http) = {
+      get : "/v1/filesystem-trim/auto-fstrim-pop"
```
Comment (talakad) [inline api/api.proto:1742]: this as well should be a put?

**Review by talakad — APPROVED** [review]
Please address the review comments

File: api/api.proto [hunk 3]
```diff
@@ -1728,48 +1728,65 @@ service OpenStorageFilesystemTrim {
         body: "*"
       };
     }
+
+  rpc AutoFSTrimPush(SdkAutoFSTrimPushRequest)
+      returns (SdkAutoFSTrimPushResponse) {
```
Comment (adityadani) [inline api/api.proto:1733]: nit: fix the indentation

File: api/api.proto [hunk 4]
```diff
@@ -1728,48 +1728,65 @@ service OpenStorageFilesystemTrim {
         body: "*"
       };
     }
+
+  rpc AutoFSTrimPush(SdkAutoFSTrimPushRequest)
```
Comment (adityadani) [inline api/api.proto:1732]: Please add a doc to indicate what this API is supposed to do.

File: api/api.proto (hunk 1 above)
Comment (adityadani) [reply api/api.proto:1735]: Not sure what this API is supposed to do, but if the API is idempotent then PUT otherwise lets use POST.

Comment (chiachunlin) [reply api/api.proto:1735]: changed to post

**Pull request merged**